src/abstract/CSBondCore.sol

```diff
@@ -216,8 +217,10 @@ abstract contract CSBondCore is ICSBondCore {
             ? amountToClaim
             : claimableShares;
         if (sharesToClaim == 0) return;
+        uint256 sharesBefore = LIDO.sharesOf(address(this));
         uint256 amount = WSTETH.wrap(_ethByShares(sharesToClaim));
-        _unsafeReduceBond(nodeOperatorId, amount);
+        uint256 sharesAfter = LIDO.sharesOf(address(this));
+        _unsafeReduceBond(nodeOperatorId, sharesBefore - sharesAfter);
 
         WSTETH.transfer(to, amount);
```

Comment:
We are transferring wstETH. The amount is precise, given it is measured before and after wrapping. The transfer amount is also correct since it is the return value of the function.